### PR TITLE
Introduce `switch` promotion test case

### PR DESCRIPTION
**Pull request by andrealinux1**
Add a new source file that introduces the test for the dual case `switch` promotion to `if` statements

### CODE REVIEW

File: share/revng/test/configuration/revng-qa/for-decompilation.yml [hunk 1]
```diff
@@ -27,3 +27,4 @@ sources:
     prefix: share/revng/test/tests/analysis/Decompilation/clang/
     members:
       - linked-lists.c
+      - dual-case-switch.c
```
Comment (pfez) [inline share/revng/test/configuration/revng-qa/for-decompilation.yml]: Please put this under `x86-64` (like e.g. `segments-and-sections.c`), not  `clang`, unless this is strictly necessary. And if for some reason you really really need `clang` I'd like you to explain why.

File: share/revng/test/tests/analysis/Decompilation/clang/dual-case-switch.c [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+LABEL_EXIT_2:
+  printf("Exit 2");
+
+LABEL_EXIT_COMMON:
+  return return_value;
+}
+
+int main() {
+  int a = 0;
```
Comment (pfez) [inline share/revng/test/tests/analysis/Decompilation/clang/dual-case-switch.c]: Is this needed?

File: share/revng/test/tests/analysis/Decompilation/clang/dual-case-switch.c [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+ * This file is distributed under the MIT License. See LICENSE.md for details.
+ */
+#include <stdio.h>
+
+int f(int arg1) {
+  int return_value;
+  while (arg1 > 0) {
+    if (arg1 == 1) {
+      return_value = 10;
+      printf("Argument equal to 1");
```
Comment (pfez) [inline share/revng/test/tests/analysis/Decompilation/clang/dual-case-switch.c]: This should be linked with `nostlib`, so no `printf` please.

I suppose this is just a way to prevent the optimizer to optimize away this call.

Just look into the other files in `x86-64` `for-decompilation` to see how we deal with this.

TLDR: use `__attribute__((weak))` all over all the functions whose call sites you want to prevent the optimizer to mess with or remove.

**Review by pfez — CHANGES_REQUESTED** [review]

File: share/revng/test/configuration/revng-qa/for-decompilation.yml (hunk 1 above)
Comment (andrealinux1) [reply share/revng/test/configuration/revng-qa/for-decompilation.yml]: No specific reason, just thought we wanted it done with `clang` since I usually do that when crafting tests. Put it under the `x86-64` folder in https://github.com/revng/revng-qa/commit/7d9d32d70292d147d1eec204ac75342c74c3fd68

File: share/revng/test/tests/analysis/Decompilation/clang/dual-case-switch.c (hunk 2 above)
Comment (andrealinux1) [reply share/revng/test/tests/analysis/Decompilation/clang/dual-case-switch.c]: Nope, it was a leftover from a previous implementation. Removed this in https://github.com/revng/revng-qa/commit/7d9d32d70292d147d1eec204ac75342c74c3fd68

File: share/revng/test/tests/analysis/Decompilation/clang/dual-case-switch.c (hunk 3 above)
Comment (andrealinux1) [reply share/revng/test/tests/analysis/Decompilation/clang/dual-case-switch.c]: Yeah right, the `printf` was there for the reason you highlighted. Changed it following your suggestion in https://github.com/revng/revng-qa/commit/7d9d32d70292d147d1eec204ac75342c74c3fd68

File: share/revng/test/tests/analysis/Decompilation/x86-64/dual-case-switch.c [hunk 4]
```diff
@@ -0,0 +1,37 @@
+/*
+ * This file is distributed under the MIT License. See LICENSE.md for details.
+ */
+
+#define WEAK __attribute__((weak))
+
+WEAK void report(int code) {
+}
+
+int f(int arg1) {
```
Comment (pfez) [inline share/revng/test/tests/analysis/Decompilation/x86-64/dual-case-switch.c]: Now I wonder why `f` is the only one not marked `WEAK`. Any specific reason?

Comment (andrealinux1) [reply share/revng/test/tests/analysis/Decompilation/x86-64/dual-case-switch.c]: No specific reason except my stupidity:) Fixed this in 91cd7fca9e648fe8c5b9286dc3909ed9cf9b83b7
